docs/reference/commandline/pull.md

```diff
@@ -24,31 +24,195 @@ Most of your images will be created on top of a base image from the
-For example, if you have set up a local registry, you can specify its path to
-pull from it. A repository path is similar to a URL, but does not contain
-a protocol specifier (`https://`, for example).
-
 To download a particular image, or set of images (i.e., a repository),
-use `docker pull`:
-
-    $ docker pull debian
-    # will pull the debian:latest image and its intermediate layers
-    $ docker pull debian:testing
-    # will pull the image named debian:testing and any intermediate
-    # layers it is based on.
-    $ docker pull debian@sha256:cbbf2f9a99b47fc460d422812b6a5adff7dfee951d8fa2e4a98caa0382cfbdbf
-    # will pull the image from the debian repository with the digest
-    # sha256:cbbf2f9a99b47fc460d422812b6a5adff7dfee951d8fa2e4a98caa0382cfbdbf
-    # and any intermediate layers it is based on.
-    # (Typically the empty `scratch` image, a MAINTAINER layer,
-    # and the un-tarred base).
-    $ docker pull --all-tags centos
-    # will pull all the images from the centos repository
-    $ docker pull registry.hub.docker.com/debian
-    # manually specifies the path to the default Docker registry. This could
-    # be replaced with the path to a local registry to pull from another source.
-    # sudo docker pull myhub.com:8080/test-image
+use `docker pull`.
+
+## Examples
+
+### Pull an image from Docker Hub
+
+To download a particular image, or set of images (i.e., a repository), use
+`docker pull`. If no tag is provided, Docker uses the `:latest` tag as a default.
+This command pulls the `debian:latest` image:
+
+```bash
+$ docker pull debian
+
+Using default tag: latest
+latest: Pulling from library/debian
+fdd5d7827f33: Pull complete
+a3ed95caeb02: Pull complete
+Digest: sha256:e7d38b3517548a1c71e41bffe9c8ae6d6d29546ce46bf62159837aad072c90aa
+Status: Downloaded newer image for debian:latest
+```
+
+Docker images can consist of multiple layers. In the example above, the image
+consists of two layers; `fdd5d7827f33` and `a3ed95caeb02`.
+
+Layers can be reused by images. For example, the `debian:jessie` image shares
+both layers with `debian:latest`. Pulling the `debian:jessie` image will therefore
+only pull its meta-data, but not its layers, because all layers are already
+present locally:
+
+```bash
+$ docker pull debian:jessie
+
+jessie: Pulling from library/debian
+fdd5d7827f33: Already exists
+a3ed95caeb02: Already exists
+Digest: sha256:a9c958be96d7d40df920e7041608f2f017af81800ca5ad23e327bc402626b58e
+Status: Downloaded newer image for debian:jessie
+```
+
+To see which images are present locally, use the [`docker images`](images.md)
+command:
+
+```bash
+$ docker images
+
+REPOSITORY   TAG      IMAGE ID        CREATED      SIZE
+debian       jessie   f50f9524513f    5 days ago   125.1 MB
+debian       latest   f50f9524513f    5 days ago   125.1 MB
+```
+
+Docker uses a content-addressable image store, and the image ID is a sha256-digest
+of the image layers. In the example above, `debian:jessie` and `debian:latest`
```

Comment:
"Docker uses a content-addressable image store, and the image ID is SHA256 digest covering the image's configuration and layers."